src/paging.rs

```diff
@@ -0,0 +1,8 @@
+/// General paging related code
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum PagetableError {
+	#[error("The accessed virtual address is not mapped")]
+	InvalidAddress,
```

Comment:
Nit: According to the description, would `NotMapped` be more fitting?

Reply:
Ah, this is just moved.

Reply:
Let's ignore this error until it actually occurs somewhere, or we differentiate between these cases.